lib/AST/Attr.cpp

```diff
@@ -1279,6 +1279,18 @@ void DifferentiableAttr::setRequirements(ASTContext &context,
   Requirements = context.AllocateCopy(requirements);
 }
 
+void DifferentiableAttr::setJVPFunction(FuncDecl *decl) {
+  JVPFunction = decl;
+  if (decl)
+    JVP = {decl->getFullName(), DeclNameLoc(decl->getNameLoc())};
```

Comment:
Using the location of `decl`'s name here (`DeclNameLoc(decl->getNameLoc())`) might not make sense.